pkg/provider/azure_vmss_cache.go

```diff
@@ -47,12 +47,23 @@ type vmssEntry struct {
 	lastUpdate    time.Time
 }
 
-type availabilitySetNodeEntry struct {
-	vmNames   sets.String
-	nodeNames sets.String
-	vms       []compute.VirtualMachine
+type nonVmssUniformNodesEntry struct {
+	vmssFlexVMNodeNames   sets.String
+	vmssFlexVMProviderIDs sets.String
+	avSetVMNodeNames      sets.String
+	avSetVMProviderIDs    sets.String
+	clusterNodeNames      sets.String
 }
 
+type VMManagementType string
+
+const (
+	ManagedByVmssUniform  VMManagementType = "ManagedByVmssUniform"
+	ManagedByVmssFlex     VMManagementType = "ManagedByVmssFlex"
+	ManagedByAvSet        VMManagementType = "ManagedByAvSet"
+	ManagedByUnknownVMSet VMManagementType = "ManagedByUnknownVMSet"
```

Comment:
should we call this "ManagedByStandalone"?

Reply:
No, both AvailabilitySet node and individual nodes will be distinguished as ManagedByAvSet, which is exactly the same as what is before this PR.

ManagedByUnknownVMSet is for handling the case where the node is neither flex nor uniform or avset. (For exmaple, in the future, we have a new node type)